src/clients/crux-wallet-onboarding-client.ts

```diff
@@ -0,0 +1,93 @@
+export interface ICruxOnBoardingClientOptions {
+    cacheStorage?: StorageService;
+    configKey?: string|IKeyManager;
+    domain?: string;
+}
+export class CruxOnBoardingClient {
+    private cacheStorage?: StorageService;
+    private initPromise: Promise<void>;
+    private cruxDomainRepository: ICruxDomainRepository;
+    private configKeyManager?: IKeyManager;
+    private cruxDomain?: CruxDomain;
+    private domainContext?: CruxDomainId;
+    constructor(options: ICruxOnBoardingClientOptions) {
+        this.cacheStorage = options.cacheStorage || new InMemStorage();
+        // set the repository constructors to be used further in the client
+        this.cruxDomainRepository = new BlockstackCruxDomainRepository({cacheStorage: this.cacheStorage});
+        // set the keyManagers if available
+        this.configKeyManager =  typeof options.configKey === "string" ? new BasicKeyManager(options.configKey) : options.configKey;
+        // set the domainContext if domain is provided
+        this.domainContext = options.domain ? new CruxDomainId(options.domain) : undefined;
+        this.initPromise = this.init();
+        log.info("CruxOnBoardingClient initialised");
+    }
+    set domain(domain: string) {
+        this.domainContext = new CruxDomainId(domain);
+    }
+    public isCruxDomainAvailable = async (domain: string): Promise<boolean> => {
+        return this.cruxDomainRepository.find(new CruxDomainId(domain));
+    }
+    public registerCruxDomain = async (domain: string): Promise<void> => {
+        // TODO: implementation of auto registration of domain on blockchain
+        throw ErrorHelper.getPackageError(null, PackageErrorCode.IsNotSupported);
+    }
+    public getCruxDomainState = async (): Promise<DomainRegistrationStatus> => {
+        await this.initPromise;
+        return this.getCruxDomain().status;
+    }
+    public getNameServiceConfig = async (): Promise<IBlockstackServiceInputOptions|undefined> => {
+        await this.initPromise;
+        return this.getCruxDomain().config.nameserviceConfiguration;
+    }
+    public getAssetMapping = async (): Promise<IClientAssetMapping> => {
+        await this.initPromise;
+        return this.getCruxDomain().config.assetMapping;
+    }
+    public putNameServiceConfig = async (newNameServiceConfig: IBlockstackServiceInputOptions): Promise<void> => {
+        await this.initPromise;
+        const cruxDomain = this.getCruxDomain();
+        cruxDomain.config.nameserviceConfiguration = newNameServiceConfig;
+        this.cruxDomain = await this.cruxDomainRepository.save(cruxDomain, this.getConfigKeyManager());
+        return;
+    }
+    public putAssetMapping = async (newAssetMapping: IClientAssetMapping): Promise<void> => {
+        await this.initPromise;
+        const cruxDomain = this.getCruxDomain();
+        cruxDomain.config.assetMapping = newAssetMapping;
+        this.cruxDomain = await this.cruxDomainRepository.save(cruxDomain, this.getConfigKeyManager());
```

Comment:
How does assetList get populated inside cruxDomain config?

Reply:
You can use `cruxDomain.setAssetMapping` - use CruxSpec.globalAssetList and populate both `config.assetMapping` and `config.assetList`

Reply:
right now, this responsibility is with the BlockstackCruxDomainRepository, I can shift it to the client if required.